packages/opentelemetry-core/src/internal/validators.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const KEY_MAX_SIZE = 256;
+const VALUE_MAX_SIZE = 256;
+
+/**
+ * Key is opaque string up to 256 characters printable. It MUST begin with a
+ * lowercase letter, and can only contain lowercase letters a-z, digits 0-9,
+ * underscores _, dashes -, asterisks *, and forward slashes /.
+ */
+export function validateKey(key: string): boolean {
+  if (key.length > KEY_MAX_SIZE || key.charAt(0) < 'a' || key.charAt(0) > 'z') {
+    return false;
+  }
+  for (let i = 1; i < key.length; i++) {
+    const c = key.charAt(i);
+    if (
```

Comment:
This condition would be a lot easier to read with regex.

Reply:
Added regex to validate the key.